server/WebAPI/Repositories/NewsRepository.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Entities.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using WebAPI.DTO;
+using Entities.DataAccess;
+using Microsoft.EntityFrameworkCore;
+
+namespace WebAPI.Repositories
+{
+    public class NewsRepository
+    {
+        private readonly FilmFinityDbContext dbContext;
+
+        public NewsRepository(FilmFinityDbContext dbContext)
+        {
+            this.dbContext = dbContext;
+        }
+
+        public IQueryable<NewsDTO> GetAllNews()
+        {
+            return dbContext.News
+                .Select(row =>
+                    new NewsDTO { 
+                        NewsId = row.NewsId, 
+                        NewsTitle = row.NewsTitle, 
+                        NewsContent = row.NewsContent,
+                        CountViews = row.CountViews,
+                        ImageSource = row.ImageSource,
+                        PublishTime = row.PublishTime,
+                        Author = new AuthorDTO { AuthorId = row.Author.AuthorId, AuthorName = row.Author.AuthorName },
+                        CelebrityJobTitles = row.NewsCategories.Select(j => j.Category)
+                            .Select(j => new CategoryDTO { CategoryId = j.CategoryId, CategoryName = j.CategoryName }).ToList()
```

Comment:
no, data base layer knows just db entities

mapping should be via automapper and at interaction layer (or interface) - in controller or in services, not at repositories

Reply:
```
[HttpGet("{id}")]
public async Task<ActionResult<MusicResource>> GetMusicById(int id)
{
    var music = await _musicService.GetMusicById(id);
    var musicResource = _mapper.Map<Music, MusicResource>(music);

    return Ok(musicResource);
}
```

Reply:
```
public class TripService
  {
    private readonly IMapper _mapper;
    private readonly TripRepository _tripRepository;

    public TripService(IMapper mapper, TripRepository tripRepository)
    {
      _mapper = mapper;
      _tripRepository = tripRepository;
    }

    public IEnumerable<TripDTO> GetTripsHistory(int userId, string searchQuery)
    {
      var dateNow = DateTime.Now;
      var trips = _tripRepository.GetUserTrips(userId)
                                 .Where(t => t.StartDate < dateNow);

      if (!string.IsNullOrEmpty(searchQuery))
        trips = trips.Where(t => t.DepartureCity.ToLower().Contains(searchQuery.ToLower()) ||
                                 t.ArrivalCountry.Name.ToLower().Contains(searchQuery.ToLower()) ||
                                 t.ArrivalCountry.NameRu.ToLower().Contains(searchQuery.ToLower()) ||
                                 t.ArrivalCountry.NameUa.ToLower().Contains(searchQuery.ToLower()));

      return _mapper.Map<IEnumerable<TripDTO>>(trips.OrderBy(t => t.StartDate).ToList());
    }
  }

```

Reply:
in our case - it is over-engineering to create special resources models for services - so, we can use second option and make mapping database entities into dto in services via automapper

Reply:
Done. Entities and DTO have been separated